bundles/org.openhab.binding.ipp/src/main/java/org/openhab/binding/ipp/internal/handler/IppPrinterHandler.java

```diff
@@ -46,73 +49,75 @@
  *
  * @author Tobias Braeutigam - Initial contribution
  */
+@NonNullByDefault
 public class IppPrinterHandler extends BaseThingHandler implements DiscoveryListener {
 
     private final Logger logger = LoggerFactory.getLogger(IppPrinterHandler.class);
 
-    private URL url;
-    private String name;
-    private CupsPrinter printer;
+    private @Nullable URL url;
+    private @Nullable CupsPrinter printer;
 
-    private int refresh = 60; // refresh every minute as default
-    ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> refreshJob;
 
-    private DiscoveryServiceRegistry discoveryServiceRegistry;
+    private final DiscoveryServiceRegistry discoveryServiceRegistry;
 
     public IppPrinterHandler(Thing thing, DiscoveryServiceRegistry discoveryServiceRegistry) {
         super(thing);
-        if (discoveryServiceRegistry != null) {
-            this.discoveryServiceRegistry = discoveryServiceRegistry;
-        }
+        this.discoveryServiceRegistry = discoveryServiceRegistry;
     }
 
     @Override
     public void initialize() {
         Configuration config = getThing().getConfiguration();
+        String name = (String) config.get(PRINTER_PARAMETER_NAME);
         try {
-            Object obj = config.get(IppBindingConstants.PRINTER_PARAMETER_URL);
-            name = (String) config.get(IppBindingConstants.PRINTER_PARAMETER_NAME);
-            if (config.get(IppBindingConstants.PRINTER_PARAMETER_REFRESH_INTERVAL) != null) {
-                BigDecimal ref = (BigDecimal) config.get(IppBindingConstants.PRINTER_PARAMETER_REFRESH_INTERVAL);
-                refresh = ref.intValue();
-            }
+            Object obj = config.get(PRINTER_PARAMETER_URL);
             if (obj instanceof URL) {
                 url = (URL) obj;
             } else if (obj instanceof String) {
                 url = new URL((String) obj);
             }
             printer = new CupsPrinter(url, name, false);
         } catch (MalformedURLException e) {
-            logger.error("malformed url {}, printer thing creation failed",
-                    config.get(IppBindingConstants.PRINTER_PARAMETER_URL));
+            logger.error("malformed url {}, printer thing creation failed", config.get(PRINTER_PARAMETER_URL));
         }
-        // until we get an update put the Thing offline
-        updateStatus(ThingStatus.OFFLINE);
-        deviceOnlineWatchdog();
-        if (this.discoveryServiceRegistry != null) {
-            this.discoveryServiceRegistry.addDiscoveryListener(this);
+
+        int refresh = 60; // refresh every minute as default
```

Comment:
Consider if worth extracting this value to a constant with descriptive name and converting code comment to javadoc for that constant.

Reply:
Done